### ISSUE TITLE
Poor accessibility of target selector

### ISSUE THREAD

**Issue by steverep** [home-assistant/frontend#14367]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

The target picker component is difficult and confusing to use with a screen reader.
- Various controls are not programmatically grouped, making it hard to separate from other inputs
- Focus is not managed well (e.g. after adding an item it's confusing what happens)
- Expand and remove buttons have double button roles, and clicking expand sometimes removes the item
- I've also experienced issues with the choose entity button appearing to be missing or not labelled correctly (possibly same problem as #14049).

### Describe the behavior you expected

- Various controls should be grouped and labelled using ARIA
- Focus should remain on a given control and move to a logical spot if necessary (e.g. sub-picker no longer in DOM)
- Only one role per interactive component
- Reliable and intuitive experience using tab and shift+tab

### Steps to reproduce the issue

n/a

### What version of Home Assistant Core has the issue?

2022.11.x

### In which browser are you experiencing the issue with?

Firefox or Chrome (latest)

### Which operating system are you using to run this browser?

Windows 10

**Comment by steverep** [comment]
Still an issue

**Comment by steverep** [comment]
Still an issue

**Comment by steverep** [comment]
Still needs work
